st3/lsp_utils/npm_client_handler.py

```diff
@@ -94,6 +94,10 @@ def additional_variables(cls) -> Optional[Dict[str, str]]:
     def minimum_node_version(cls) -> Tuple[int, int, int]:
         return (8, 0, 0)
 
+    @classmethod
+    def install_in_cache(cls) -> bool:
+        return True
```

Comment:
Let's also add a `basedir()` class method to AbstractPlugin?

Reply:
You mean to define where it gets installed?

Reply:
Yes :)

Reply:
Then the method name should probably be something more descriptive like "install_basedir()".

But I don't really get why should that be on the plugin interface since it's the plugin itself that handles installation and it's nothing that LSP (package) itself is concerned about.

Reply:
There are more plugins than just those based on npm.

Reply:
Yes but why would that path be defined on the AbstractPlugin interface. The installation is entirely handled by the plugin (regardless if npm-based or not).

Reply:
I think it's good to be `False` by default?

Reply:
It will be false once all existing packages are migrated to new structure. For now it's opt in so that on upgrading lsp-* packages, the old directory is cleaned up.

But now I'm wondering if upgrading dependency is maybe disabling all packages that depend on it... In that case we could set it to false straight away.

Reply:
I've asked on Discord and was told that updating dependency doesn't disable all packages that depend on it so unfortunately, we need to handle that per LSP package.